### PR TITLE
dep: update tidb to introduce some change

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #4287 

### What is changed and how it works?

introduce https://github.com/pingcap/tidb/pull/35394 , when downstream TiDB is updated, we now align with the error code. For the old behaviour of TiDB, we make use of auto-resume

#### Tests 

 - Unit test
 - Integration test
   this is another fix to improve the stability of concurrent modify column by two workers, like https://github.com/pingcap/tiflow/blob/b4600f5a22cec26c17d284ff5329f6ff9bc7d8a1/dm/tests/shardddl2_1/run.sh#L477-L481

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: dm/syncer/dbconn/db.go [hunk 1]
```diff
@@ -210,8 +207,7 @@ func (conn *DBConn) ExecuteSQLWithIgnore(
 					zap.Error(err))
 				return true
 			}
-			// TODO: move it to above IsRetryableError
-			return isRetryableError(err)
+			return false
```
Comment (buchuitoudegou) [inline dm/syncer/dbconn/db.go:210]: Should `ErrColumnInChange` be retried? In https://github.com/pingcap/tidb/issues/35102 , this is supposed to be retried.

Comment (D3Hunter) [reply dm/syncer/dbconn/db.go:210]: included in https://github.com/pingcap/tiflow/blob/7974dd31256db1d5cad1199dc36cd4aa79fa7541/dm/syncer/dbconn/db.go#L114

Comment (lance6716) [reply dm/syncer/dbconn/db.go:210]: please read PR description and title

**Review by D3Hunter — COMMENTED** [review]
lgtm

File: dm/syncer/dbconn/db.go (hunk 1 above)
Comment (lance6716) [reply dm/syncer/dbconn/db.go:210]: https://github.com/pingcap/tidb/pull/35394/files#diff-dd4803f526cf56d6adca5e33b6a76193a035cd98ed31ed8e50b1698635c296c8

**Review by D3Hunter — APPROVED** [review]

**Review by buchuitoudegou — APPROVED** [review]

**Pull request merged**